src/Corax/IndexWriter.cs

```diff
@@ -299,6 +300,42 @@ public long Index(string id, Span<byte> data)
             return Index(idSlice, data);
         }
 
+        public unsafe long Index(string id, Span<byte> data, float documentBoost)
+        {
+            var entryId = Index(id, data);
+            AppendDocumentBoost(entryId, documentBoost);
+            return entryId;
+        }
+
+        private unsafe void AppendDocumentBoost(long entryId, float documentBoost)
+        {
+            _documentBoost ??= Transaction.FixedTreeFor(Constants.DocumentBoostSlice, sizeof(float));
+            
+            documentBoost = documentBoost switch
+            {
+                <= 1 => MathF.Log(2),
+                _ => MathF.Log(documentBoost)
+            };
+            
+            var buffer = stackalloc byte[sizeof(float)];
+            Unsafe.WriteUnaligned(buffer, documentBoost);
+            using var _ = Slice.From(Transaction.Allocator, buffer, sizeof(float), out var floatSlice);
+            _documentBoost.Add(entryId, floatSlice);
+        }
+
+        private unsafe void RemoveDocumentBoost(long entryId)
+        {
+            _documentBoost ??= Transaction.FixedTreeFor(Constants.DocumentBoostSlice, sizeof(float));
+            _documentBoost.Delete(entryId);
+        }
+
+        public unsafe long Update(string field, Span<byte> key, LazyStringValue id, Span<byte> data, ref long numberOfEntries, float documentBoost)
+        {
+            var entryId = Update(field, key, id, data, ref numberOfEntries);
+            AppendDocumentBoost(entryId, documentBoost);
```

Comment:
here, if we skip on `documentBooost` of 1, we may need to check if the boost value changed.